src/init.cpp

```diff
@@ -814,14 +814,35 @@ static std::string ResolveErrMsg(const char * const optname, const std::string&
     return strprintf(_("Cannot resolve -%s address: '%s'"), optname, strBind);
 }
 
+/**
+ * Used when -printtoconsole is not explicitly set to determine if it makes
+ * sense to log messages to stdout.
+ *
+ * @return true if we should log to stdout, false otherwise.
+ */
+static bool ShouldPrintToConsole()
```

Comment:
nit: `PrintToConsoleDefault()` is seems more accurate. Otherwise the function name makes it sound like it takes `-printtoconsole` into account.

Reply:
Since it is only called once, you could also inline it and call the variable `const bool default_printtoconsole = !gArgs.GetBoolArg("-daemon", false);`

Reply:
It's already inlined by the compiler, there's no difference in the executable produced by GCC either way. The only reason it's still a method is because it originally called `isatty()`. If I can get #12713 merged I am happy to remove this method.